src/markdown.ts

```diff
@@ -316,7 +318,7 @@ function renderIntoPieces(renderer: Renderer, root: string, sourcePath: string):
     }
     let result = "";
     for (const piece of context.pieces) {
-      result += piece.html = normalizePieceHtml(piece.html, sourcePath, context);
+      result += piece.html = normalizePieceHtml(piece.html, root, sourcePath, context);
```

Comment:
If we must introduce it, could `root` be part of the `context`?

Reply:
done.  this cleaned up the code a bunch thank you!